app/controllers/reportings_controller.rb

```diff
@@ -35,6 +35,11 @@ class ReportingsController < ApplicationController
   before_filter :find_project_by_project_id
   before_filter :authorize
 
+  before_filter :find_reporting, only: [:show, :edit, :update, :confirm_destroy, :destroy]
+  before_filter :build_reporting, only: :create
+
+  before_filter :check_visibility, except: [:create, :index, :new, :available_projects]
```

Comment:
After thinking about it, I see why there should be no visibility check for create and new, because there is nothing that could not be **visible**.

Though I am quite sure that there is some kind of authorization check missing (e.g. are you even allowed to create status reportings?)